src/plugins/transform-jsx-spread/index.js

```diff
@@ -28,33 +77,32 @@ export default ({ types: t }) => {
       }
     }
 
-    if (t.isJSXNamespacedName(node.name)) {
-      node.name = t.stringLiteral(
-        node.name.namespace.name + ":" + node.name.name.name
-      );
-    } else if (esutils.keyword.isIdentifierNameES6(node.name.name)) {
-      node.name.type = "Identifier";
-    } else {
-      node.name = t.stringLiteral(node.name.name);
-    }
-
-    return t.inherits(t.objectProperty(node.name, value), node);
+    return value;
   }
 
   return {
-    name: "transform-hoist-tagged-templates",
+    name: "transform-jsx-spread",
     visitor: {
-      JSXOpeningElement(path) {
+      JSXOpeningElement(path, state) {
+        const useSpread = state.opts.useSpread === true;
         const hasSpread = path.node.attributes.some(t.isJSXSpreadAttribute);
```

Comment:
While it is not strictly related to this PR, this would be better:
```suggestion
        const hasSpread = path.node.attributes.some(attr => t.isJSXSpreadAttribute(attr));
```
That's because `t.is*` functions accept a second parameter of properties to check on the node, and `some` passes the array index.

Reply:
Ah good catch. I've been fixing these as I find them, must have missed this one.